src/coreclr/vm/corhost.cpp

```diff
@@ -305,6 +307,11 @@ HRESULT CorHost2::ExecuteAssembly(DWORD dwAppDomainId,
         }
     }
 
+    {
+        GCX_COOP();
+        unhandledExceptionMarkerFrame.Push(pThread);
+    }
+
     INSTALL_UNHANDLED_MANAGED_EXCEPTION_TRAP;
```

Comment:
Should the unhandledExceptionMarkerFrame be coupled with `INSTALL_UNHANDLED_MANAGED_EXCEPTION_TRAP`? They seem to be handling same scenario.